server/lyft/gateway/events_controller.go

```diff
@@ -0,0 +1,287 @@
+	if err != nil {
+		wrapped := errors.Wrapf(err, "Error parsing pull data: %s %s", err, githubReqID)
+		return HttpResponse{
+			body: wrapped.Error(),
+			err: HttpError{
+				code: http.StatusBadRequest,
+				err:  wrapped,
+			},
+		}
+	}
+	logger.Debug("identified event as type %q", pullEventType.String())
+	return g.handlePullRequestEvent(baseRepo, headRepo, pull, user, pullEventType, r)
+}
+
+func (g *VCSEventsController) handlePullRequestEvent(baseRepo models.Repo, headRepo models.Repo, pull models.PullRequest, user models.User, eventType models.PullRequestEventType, request *http.Request) HttpResponse {
+	if !g.RepoAllowlistChecker.IsAllowlisted(baseRepo.FullName, baseRepo.VCSHost.Hostname) {
+		// If the repo isn't allowlisted and we receive an opened pull request
+		// event we comment back on the pull request that the repo isn't
+		// allowlisted. This is because the user might be expecting Atlantis to
+		// autoplan. For other events, we just ignore them.
+		if eventType == models.OpenedPullEvent {
+			g.commentNotAllowlisted(baseRepo, pull.Num)
+		}
+		err := errors.New(fmt.Sprintf("Pull request event from non-allowlisted repo \"%s/%s\"", baseRepo.VCSHost.Hostname, baseRepo.FullName))
+		return HttpResponse{
+			body: err.Error(),
+			err: HttpError{
+				code: http.StatusForbidden,
+				err:  err,
+			},
+		}
+	}
+	switch eventType {
+	case models.OpenedPullEvent, models.UpdatedPullEvent:
+		// If the pull request was opened or updated, we perform a pseudo-autoplan to determine if tf changes exist.
+		// If it exists, then we will forward request to the worker.
+		go g.handleOpenPullEvent(baseRepo, headRepo, pull, user, request)
+		return HttpResponse{
+			body: "Processing...",
+		}
+	case models.ClosedPullEvent:
+		// If the pull request was closed, we route to worker to handle deleting locks.
+		if err := g.SendToWorker(request); err != nil {
+			return HttpResponse{
+				body: err.Error(),
+				err: HttpError{
+					code: http.StatusBadRequest,
+					err:  err,
+				},
+			}
+		}
+	case models.OtherPullEvent:
+		// Else we ignore the event.
+		return HttpResponse{
+			body: "Ignoring non-actionable pull request event",
+		}
+	}
+	return HttpResponse{}
+}
+
+func (g *VCSEventsController) handleOpenPullEvent(baseRepo models.Repo, headRepo models.Repo, pull models.PullRequest, user models.User, request *http.Request) {
+	if hasTerraformChanges := g.AutoplanValidator.PullRequestHasTerraformChanges(baseRepo, headRepo, pull, user); hasTerraformChanges {
+		if err := g.SendToWorker(request); err != nil {
+			g.Logger.With("err", err).Warn("failed to send autoplan request to Atlantis worker")
+		}
+	}
+}
+
+func (g *VCSEventsController) SendToWorker(r *http.Request) error {
+	buffer := bytes.NewBuffer([]byte{})
+	if err := r.Write(buffer); err != nil {
+		g.Scope.SubScope("send").Counter("failure").Inc(1)
+		err = errors.Wrap(err, "marshalling gateway request to buffer")
+		return err
```

Comment:
```suggestion
		return errors.Wrap(err, "marshalling gateway request to buffer")
```

same goes for the rest.